.circleci/config.yml

```diff
@@ -2,23 +2,54 @@
 version: 2.1
 
 orbs:
+  # Code Coverage
   codecov: codecov/codecov@1.0.5
+  # For Installing Browsers
   browser-tools: circleci/browser-tools@0.1.3
+  # For Testing on windows
+  win: circleci/windows@2.4.0
 
 defaults: &defaults
+  working_directory: ~/lightning-language-server
   docker:
     - image: circleci/node:12.15.0
 
 jobs:
-  build:
+  # Default Job
+  build-and-test:
     <<: *defaults
+    description: "Lint, Build, Test with Coverage"
     steps:
       - checkout
       - restore_cache:
           key: dependency-cache-{{ checksum "package.json" }}
+
+      # Build and Lint
       - run: yarn install
       - run: yarn lint
+
+      # Cache the node_modules folder so if package.json doesn't change
+      # then we can prevent downloading it all over again.
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - node_modules
+
+      # Save the current workspace state so the next jobs can
+      # re-use it without having to do the build over again.
+      - persist_to_workspace:
+          root: ~/lightning-language-server
+          paths:
```

Comment:
might be a maintenance annoyance going forward. how much longer is it really to have the other job rebuild only?

Reply:
You mean they should all just checkout, install, build before doing their work?

Reply:
that just seems simpler than having to remember to go here and add new paths if we add a new folder or file later. if it's notably faster though then I'm ok with it

Reply:
It'll be faster, we also shouldn't need any new stuff in here. Most things should be in the packages directory.